swift-format still can't treat `unsafe` expressions correctly in some cases

Thanks for fixing #972. However, it seems that swift-format still can't treat `unsafe` expressions correctly in some cases. The cases are when a l-value is unsafe.

## Input

```swift
let ptr: UnsafePointer<UInt8>?
// comment
unsafe ptr = .init(bitPattern: 0)
print(ptr)
```

## Expected result

```swift
let ptr: UnsafePointer<UInt8>?
// comment
unsafe ptr = .init(bitPattern: 0)
print(ptr)
```

## Actual result

```swift
let ptr: UnsafePointer<UInt8>?
// comment

// comment
unsafe ptr = .init(bitPattern: 0)
print(ptr)
```

## Environment

- swift-format: 58807b9061310e626e8be9a2831be2fb7ef99858
- swift-syntax: [820501e3cf5894ad1deb8de95649b938a8bbe9f4](https://github.com/swiftlang/swift-syntax/commit/820501e3cf5894ad1deb8de95649b938a8bbe9f4)

I've confirmed that swift-format's format mode have been fixed thanks to swiftlang/swift-syntax#3043.

However, its lint mode is still broken. Here is how to reproduce the issue.

## Input (test.swift)

```swift
let ptr: UnsafePointer<UInt8>?
// comment
unsafe ptr = .init(bitPattern: 0)
print(ptr)
```

## `.swift-format`

```json
{
  "rules" : {
    "NoAssignmentInExpressions" : true
  },
  "version" : 1
}
```

## Steps to reproduce

1. Create `test.swift`
2. Create `.swift-format`
3. Run `swift format lint test.swift`

## Expected result

(no warnings)

## Actual result

```
test.swift:3:8: warning: [NoAssignmentInExpressions] move this assignment expression into its own statement
test.swift:3:8: warning: [NoAssignmentInExpressions] move this assignment expression into its own statement
```

--- a/Sources/SwiftFormat/Rules/NoAssignmentInExpressions.swift
+++ b/Sources/SwiftFormat/Rules/NoAssignmentInExpressions.swift
@@ -113,15 +113,15 @@ public final class NoAssignmentInExpressions: SyntaxFormatRule {
 
   /// Returns a value indicating whether the given node is a standalone assignment statement.
   ///
-  /// This function considers try/await expressions and automatically walks up through them as
-  /// needed. This is because `try f().x = y` should still be a standalone assignment for our
+  /// This function considers try/await/unsafe expressions and automatically walks up through them
+  /// as needed. This is because `try f().x = y` should still be a standalone assignment for our
   /// purposes, even though a `TryExpr` will wrap the `InfixOperatorExpr` and thus would not be
   /// considered a standalone assignment if we only checked the infix expression for a
   /// `CodeBlockItem` parent.
   private func isStandaloneAssignmentStatement(_ node: InfixOperatorExprSyntax) -> Bool {
     var node = Syntax(node)
     while let parent = node.parent,
-      parent.is(TryExprSyntax.self) || parent.is(AwaitExprSyntax.self)
+      parent.is(TryExprSyntax.self) || parent.is(AwaitExprSyntax.self) || parent.is(UnsafeExprSyntax.self)
     {
       node = parent
     }

--- a/Tests/SwiftFormatTests/Rules/NoAssignmentInExpressionsTests.swift
+++ b/Tests/SwiftFormatTests/Rules/NoAssignmentInExpressionsTests.swift
@@ -187,19 +187,21 @@ final class NoAssignmentInExpressionsTests: LintOrFormatRuleTestCase {
     )
   }
 
-  func testTryAndAwaitAssignmentExpressionsAreUnchanged() {
+  func testTryAndAwaitAndUnsafeAssignmentExpressionsAreUnchanged() {
     assertFormatting(
       NoAssignmentInExpressions.self,
       input: """
         func foo() {
           try a.b = c
           await a.b = c
+          unsafe a.b = c
         }
         """,
       expected: """
         func foo() {
           try a.b = c
           await a.b = c
+          unsafe a.b = c
         }
         """,
       findings: []

